src/main/kotlin/app/revanced/patches/youtube/misc/fix/backtoexit/patch/BackToExitPatch.kt

```diff
@@ -0,0 +1,62 @@
+package app.revanced.patches.youtube.misc.fix.backtoexit.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstruction
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint.Companion.resolve
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patches.youtube.misc.fix.backtoexit.annotation.BackToExitCompatibility
+import app.revanced.patches.youtube.misc.fix.backtoexit.fingerprints.OnBackPressedFingerprint
+import app.revanced.patches.youtube.misc.fix.backtoexit.fingerprints.RecyclerViewScrollingFingerprint
+import app.revanced.patches.youtube.misc.fix.backtoexit.fingerprints.RecyclerViewTopScrollingParentFingerprint
+import app.revanced.patches.youtube.misc.fix.backtoexit.fingerprints.RecyclerViewTopScrollingFingerprint
+
+@Description("Close the app by tapping the back button from the home feed.")
+@BackToExitCompatibility
+@Version("0.0.1")
+class BackToExitPatch : BytecodePatch(
+    listOf(
+        OnBackPressedFingerprint,
+        RecyclerViewScrollingFingerprint,
+        RecyclerViewTopScrollingParentFingerprint,
+    )
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        with (RecyclerViewScrollingFingerprint.result!!) {
+            mutableMethod.addInstruction(
+                scanResult.patternScanResult!!.endIndex,
+                invokeString("", "onStartScrollView", "")
+            )
+        }
+
+        RecyclerViewTopScrollingFingerprint.resolve(context, RecyclerViewTopScrollingParentFingerprint.result!!.classDef)
+
+        with (RecyclerViewTopScrollingFingerprint.result!!) {
+            mutableMethod.addInstruction(
+                scanResult.patternScanResult!!.endIndex + 1,
+                invokeString("", "onStopScrollView", "")
+            )
+        }
+
+        with (OnBackPressedFingerprint.result!!) {
+            mutableMethod.addInstruction(
+                scanResult.patternScanResult!!.endIndex,
+                invokeString("p0", "exitOnBackPressed", "Lcom/google/android/apps/youtube/app/watchwhile/WatchWhileActivity;")
+            )
+        }
+
+        return PatchResultSuccess()
+    }
+
+    private companion object {
+        fun invokeString(
```

Comment:
This method should not simply build a string. It should be called `injectCall` and accept a method to add the instruction to, index to add the instruction, and the remaining relevant current parameters via a class.